main.go

```diff
@@ -396,6 +486,177 @@ func runInPod(name string, cmdToRun string, env []string, timeout int) ([]string
+	var selection []int
+	if step.Selection.Range.Order == SEQUENTIAL {
+	  if step.Selection.Range.Start <= 0 || 
+			step.Selection.Range.End - step.Selection.Range.Start + 1 > config.Nodes {
+				fatal("Invalid range")
+		}
+		selection = makeRange(step.Selection.Range.Start, step.Selection.Range.End)
+	} else if step.Selection.Range.Order == RANDOM {
+		if step.Selection.Range.Number > config.Nodes {
+			fatal("Invalid range")
+		}
+		selection = onePerm(config.Nodes)[0:step.Selection.Range.Number]
+
+	} else /* Invalid order */ {
+			fatal("Invalid Selection format. Order must be SEQUENTIAL or RANDOM ")
+	}
+	return selection
+} 
+
+func selectNodesPercent(step *Step, config Config) []int {
+	var percent int
+	var selection []int
+	percent = step.Selection.Percent.Percent
+	if percent > 100 || percent < 0 {
+		fatal("Invalid percent")
+	}
+	numNodes := int( (float64(percent) / 100.0) * float64(config.Nodes) )
+	if step.Selection.Percent.Order == SEQUENTIAL {
+		if step.Selection.Percent.Start -1 + numNodes > config.Nodes {
+			fatal("Invalid start position")
+		}
+		selection = makeRange(step.Selection.Percent.Start, step.Selection.Percent.Start -1 + numNodes)
+	} else if step.Selection.Percent.Order == RANDOM {
+		selection = onePerm(config.Nodes)[0:numNodes]
+	} else /* Invalid order */ {
+		fatal("Invalid Selection format. Order must be SEQUENTIAL or RANDOM ")
+	}
+	return selection
+}
+
+func selectNodesSubset(step *Step, subsetPartition map[int][]int) []int{
+	if (subsetPartition == nil) {
+		fatal("Subset specified without specifying partion in header")
+	}
+	selected := make([]int, 0)
+	for _, partition := range step.Selection.Subset.Indices {
+		selected = append(selected, subsetPartition[partition]...)
+	}
+	return selected
+}
+
+
+func partition(config Config) map[int][]int{
+	if config.SubsetPartition == nil {
+		return nil
+	} 
+	partitionMap := make(map[int][]int)
+
+	if config.SubsetPartition.Order == SEQUENTIAL {
+		if config.SubsetPartition.PartitionType == EVEN {
+			seqEvenPartition(&partitionMap, config.SubsetPartition.NumberPartitions, config.Nodes)
+		} else if config.SubsetPartition.PartitionType == WEIGHTED {
+			seqWeightedPartition(&partitionMap, config.SubsetPartition.Percents, config.Nodes)
+		} else /* invalid partition type */{
+				fatal("Invalid partition type")
+		}
+	} else if config.SubsetPartition.Order == RANDOM {
+		if config.SubsetPartition.PartitionType == EVEN {
+			randEvenPartition(&partitionMap, config.SubsetPartition.NumberPartitions, config.Nodes)
+		} else if config.SubsetPartition.PartitionType == WEIGHTED {
+			randWeightedPartition(&partitionMap, config.SubsetPartition.Percents, config.Nodes)
+		} else /* invalid partition type */{
+				fatal("Invalid partition type")
+		}
+	} else { /* invalid ordering */
+		fatal("Partition has invalid order")
+	}
+	return partitionMap
+}
+
+func seqEvenPartition(partitionMap *map[int][]int, numSubsets int, numNodes int) {
```

Comment:
I don't think your map needs to be a pointer.

Go maps are pointers themselves, so it's not like your having to copy the whole map when you pass it as a parameter (citation needed, I'm not 100% sure but reasonably sure).